src/main/resources/specification/specification.yaml

```diff
@@ -10,4 +10,974 @@ info:
+          description: Unauthorized
+        '403':
+          description: Forbidden
+  /catalogs/{catalogId}/datasets/{id}:
+    get:
+      security:
+        - bearerAuth: [ ]
+      tags:
+        - dataset
+      summary: Get a specific dataset from specififed catalog
+      description: Get a specific dataset specififed catalog
+      operationId: getDatasetById
+      parameters:
+        - name: catalogId
+          in: path
+          description: catalog id
+          required: true
+          schema:
+            type: string
+        - name: id
+          in: path
+          description: dataset id
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Dataset"
+        '401':
+          description: Not logged in
+        '403':
+          description: Forbidden
+        '404':
+          description: Not Found
+    patch:
+      security:
+        - bearerAuth: []
+      tags:
+        - dataset
+      summary: Update dataset in specified catalog
+      description: Update dataset in specified catalog
+      operationId: updateDataset
+      parameters:
+        - name: catalogId
+          in: path
+          description: catalog id
+          required: true
+          schema:
+            type: string
+        - name: id
+          in: path
+          description: dataset id
+          required: true
+          schema:
+            type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Dataset'
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Not logged in
+        '403':
+          description: Forbidden
+        '404':
+          description: Not Found
+    delete:
+      security:
+        - bearerAuth: [ ]
+      tags:
+        - dataset
+      summary: Delete a specific dataset
+      description: Delete a specific dataset
+      operationId: removeDataset
+      parameters:
+        - name: catalogId
+          in: path
+          description: catalog id
+          required: true
+          schema:
+            type: string
+        - name: id
+          in: path
+          description: id
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '400':
+          description: Bad Request
+        '401':
+          description: Not logged in
+        '403':
+          description: Forbidden
+components:
+  schemas:
+    Catalog:
+      type: object
+      description: Dataset catalog
+      properties:
+        id:
+          type: string
+          description: Catalog id
+        uri:
+          type: string
+          description: Unique resource identifier for catalog
+        title:
```

Comment:
For alle slike felt som har språk, kan du definere en type:

title:
          $ref: '#/components/schemas/TextField'

og f.eks. ha:
    TextField:
      type: object
      properties:
        nb:
          type: string
          description: the text in Bokmål(Norwegian)
        nn:
          type: string
          description: the text in Nynorsk(Norwegian)
        'no':
          type: string
          description: the text in Norwegian(general)
        en:
          type: string
          description: the text in Nynorsk(general)